### ISSUE TITLE
[Twitter] Randomly getting Bad guest token for the same video

### ISSUE THREAD

**Issue by someziggyman** [yt-dlp/yt-dlp#3796]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.05.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Anywhere

### Description

Test link:
https://twitter.com/BBCSteveR/status/1526329765065539592

Just try reproducing this about 5 times. one will definitely fail. If not, then try to wait for a minute or two after success and try again. Hopefully, it's reproducible. 

### Verbose log

```shell
=======
Failure log
=======
./yt-dlp -v -F "https://twitter.com/BBCSteveR/status/1526329765065539592"
[debug] Command-line config: ['-v', '-F', 'https://twitter.com/BBCSteveR/status/1526329765065539592']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355] (zip)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.9.12 (CPython 64bit) - macOS-12.3.1-arm64-arm-64bit
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: avprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: avconv -bsfs
[debug] exe versions: none
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] [twitter] Extracting URL: https://twitter.com/BBCSteveR/status/1526329765065539592
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
ERROR: [twitter] 1526329765065539592: Bad guest token.
  File "./yt-dlp/yt_dlp/extractor/common.py", line 642, in extract
    ie_result = self._real_extract(url)
  File "./yt-dlp/yt_dlp/extractor/twitter.py", line 441, in _real_extract
    status = self._call_api(
  File "./yt-dlp/yt_dlp/extractor/twitter.py", line 104, in _call_api
    raise ExtractorError(self._parse_json(



=======
success log
=======

./yt-dlp -v -F "https://twitter.com/BBCSteveR/status/1526329765065539592"
[debug] Command-line config: ['-v', '-F', 'https://twitter.com/BBCSteveR/status/1526329765065539592']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355] (zip)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.9.12 (CPython 64bit) - macOS-12.3.1-arm64-arm-64bit
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: avprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: avconv -bsfs
[debug] exe versions: none
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] [twitter] Extracting URL: https://twitter.com/BBCSteveR/status/1526329765065539592
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[debug] Sort order given by extractor: res, br, size, proto
[debug] Formats sorted by: hasvid, ie_pref, res, tbr, vbr, abr, filesize, fs_approx, proto, lang, quality, fps, hdr:12(7), vcodec:vp9.2(10), acodec, asr, vext, aext, hasaud, source, id
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
```

**Comment by pukkandan** [comment]
Can't reproduce
```
❯ repeat yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Calling: python.exe D:\Programs\Scripts\repeat.py "yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F"
Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0

Running: yt-dlp https://twitter.com/BBCSteveR/status/1526329765065539592 -F
Latest version: 2022.05.18, Current version: 2022.05.18
yt-dlp is up to date (2022.05.18)
[twitter] 1526329765065539592: Downloading guest token
[twitter] 1526329765065539592: Downloading JSON metadata
[twitter] 1526329765065539592: Downloading m3u8 information
[info] Available formats for 1526329765065539592:
ID        EXT RESOLUTION │  FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────────────
hls-288   mp4 480x270    │ ~12.91MiB  288k m3u8_n │ avc1.4d001e  288k mp4a.40.2  0k
http-288  mp4 480x270    │ ~12.91MiB  288k https  │ unknown      288k unknown    0k
hls-832   mp4 640x360    │ ~37.30MiB  832k m3u8_n │ avc1.4d001f  832k mp4a.40.2  0k
http-832  mp4 640x360    │ ~37.30MiB  832k https  │ unknown      832k unknown    0k
hls-2176  mp4 1280x720   │ ~97.56MiB 2176k m3u8_n │ avc1.640020 2176k mp4a.40.2  0k
http-2176 mp4 1280x720   │ ~97.56MiB 2176k https  │ unknown     2176k unknown    0k
Command exited with error code 0
```

**Comment by pukkandan** [comment]
No longer relevent
